taxi/tests/test_views.py

```diff
@@ -0,0 +1,194 @@
+from django.contrib.auth import get_user_model
+from django.test import TestCase, Client
+from django.urls import reverse
+
+from taxi.models import Manufacturer, Car, Driver
+
+
+class LoginRequiredTest(TestCase):
+    def setUp(self) -> None:
+        manufacturer = Manufacturer.objects.create(
+            name="test name",
+            country="test country"
+        )
+        get_user_model().objects.create_user(
+            username="test_user",
+            password="test123user"
+        )
+        Car.objects.create(
+            model="test model",
+            manufacturer=manufacturer
+        )
+
+    def test_main_pages(self) -> None:
+        urls = (
+            "taxi:index",
+            "taxi:manufacturer-list",
+            "taxi:manufacturer-create",
+            "taxi:car-list",
+            "taxi:car-create",
+            "taxi:driver-list",
+            "taxi:driver-create",
+        )
+        for url in urls:
+            response = self.client.get(reverse(url))
+            self.assertNotEqual(response.status_code, 200)
+
+    def test_pages_with_pk(self) -> None:
+        urls = (
+            "taxi:manufacturer-update",
+            "taxi:manufacturer-delete",
+            "taxi:car-detail",
+            "taxi:car-update",
+            "taxi:car-delete",
+            "taxi:driver-detail",
+            "taxi:driver-update",
+            "taxi:driver-delete",
+            "taxi:toggle-car-assign",
+        )
+        for url in urls:
+            response = self.client.get(reverse(url, args=[1]))
+            self.assertNotEqual(response.status_code, 200)

```

Comment:
In the `test_pages_with_pk` method, similar to `test_main_pages`, consider checking for a specific status code like 302 (redirect) to confirm that the user is redirected to the login page when not authenticated.